build.gradle

```diff
@@ -372,8 +399,42 @@ task copyHtml(type: Copy) {
 // TODO: Temporary workaround for new exes / JRE bundling not quite being officially ready yet
 distZip.exclude("*.x*.exe")
 
+task unpackZipForNSIS(type: Copy) {
+    dependsOn distZip
+
+    def zipFile = file("${buildDir}/distributions/TerasologyLauncher.zip")
+    def tmpDir = file("${buildDir}/tmp/distNSIS")
+    from zipTree(zipFile)
+    into tmpDir
+}
+
+task renderNSISScript(type: Copy) {
+    dependsOn unpackZipForNSIS
+
+    from("win-setup/win-setup.nsi")
+    into("${buildDir}/tmp/distNSIS")
+    expand(appName: "TerasologyLauncher", guiName: "Terasology Launcher", publisher: orgName,
+      verString: displayVersion, verMajor: versionMajor, verMinor: versionMinor,
+      stringBundlesDir: "${projectDir}/win-setup/resources/strings", outDir: "${buildDir}/distributions",
+      licenseFile: "${projectDir}/LICENSE", minJREVersion: targetCompatibility.toString())
+}
+
+task copyNSISresources(type: Copy) {
+    from("win-setup/resources/images")
+    into("${buildDir}/tmp/distNSIS")
+}
+
+task distNSIS(type: Exec) {
+    dependsOn renderNSISScript
+    dependsOn copyNSISresources
+
+    executable "makensis"
```

Comment:
Actually fails with the following message:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':distNSIS'.
> A problem occurred starting process 'command 'makensis''

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

Not sure if it's graceful enough, if it isn't please provide me with the message you'd like to see so I can improve it.